src/main/java/picard/sam/SamToFastq.java

```diff
@@ -469,17 +486,25 @@ private void assertPairedMates(final SAMRecord record1, final SAMRecord record2)
     static final class FastqWriters {
         private final FastqWriter firstOfPair, unpaired;
         private final Lazy<FastqWriter> secondOfPair;
+        private final Map<SAMReadGroupRecord, List<FastqWriter>> additionalWriters;
 
-        /** Constructor if the consumer wishes for the second-of-pair writer to be built on-the-fly. */
-        private FastqWriters(final FastqWriter firstOfPair, final Lazy<FastqWriter> secondOfPair, final FastqWriter unpaired) {
+        /**
+         * Constructor if the consumer wishes for the second-of-pair writer to be built on-the-fly.
+         */
+        private FastqWriters(final FastqWriter firstOfPair, final Lazy<FastqWriter> secondOfPair, final FastqWriter unpaired,
+                             Map<SAMReadGroupRecord, List<FastqWriter>> additionalWriters) {
             this.firstOfPair = firstOfPair;
             this.unpaired = unpaired;
             this.secondOfPair = secondOfPair;
+            this.additionalWriters = additionalWriters;
         }
 
-        /** Simple constructor; all writers are pre-initialized.. */
-        private FastqWriters(final FastqWriter firstOfPair, final FastqWriter secondOfPair, final FastqWriter unpaired) {
-            this(firstOfPair, new Lazy<>(() -> secondOfPair), unpaired);
+        /**
+         * Simple constructor; all writers are pre-initialized..
+         */
+        private FastqWriters(final FastqWriter firstOfPair, final FastqWriter secondOfPair, final FastqWriter unpaired,
+                             Map<SAMReadGroupRecord, List<FastqWriter>> additionalWriters) {
+            this(firstOfPair, new Lazy<>(() -> secondOfPair), unpaired, additionalWriters);
         }
 
         public FastqWriter getFirstOfPair() {

```

Comment:
pretty sure all the methods in this nested class can be `private`

Reply:
done